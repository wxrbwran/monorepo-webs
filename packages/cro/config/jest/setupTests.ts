// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom/extend-expect';

jest.mock('@/services/auth', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
  };
  return {
    ajax: API,
    auth: API,
  };
});

jest.mock('@/services/http', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    setAuthorizationToken: jest.fn(),
  };
  const http = API;
  return http;
});
