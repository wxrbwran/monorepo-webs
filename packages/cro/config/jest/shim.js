import 'core-js/stable';
import 'regenerator-runtime/runtime';

/* eslint-disable import/first */
global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function requestAnimationFrame(callback) {
    setTimeout(callback, 0);
  };

global.matchMedia = window.matchMedia = () => ({
  matches: false,
  addListener: () => { },
  removeListener: () => { },
});
