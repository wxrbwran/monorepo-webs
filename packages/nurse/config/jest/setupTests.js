// import Enzyme from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
import 'jest';
// import axios from 'axios';
// import '@testing-library/jest-dom/extend-expect';

// Enzyme.configure({ adapter: new Adapter() });

jest.mock('@/services/auth', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
  };
  return {
    ajax: API,
    auth: API,
  };
});

jest.mock('@/services/http', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    setAuthorizationToken: jest.fn(),
  };
  const http = API;
  return http;
});

// jest.mock('@/services/api', () => {
//     const API = {
//         get: jest.fn(() => Promise.resolve({
//             data: {}, status: 'success',
//         })),
//         post: jest.fn(() => Promise.resolve({
//             data: {}, status: 'success',
//         })),
//         setAuthorizationToken: jest.fn(),
//     };
//     const http = API;
//     return http;
// })
