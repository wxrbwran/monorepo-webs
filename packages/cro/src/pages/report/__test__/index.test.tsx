import 'jest';
import React from 'react';
import Report from '..';
import { render, RenderResult } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
let wrapper: RenderResult;

describe('报告页面', () => {
  beforeEach(() => {
    wrapper = render(<Router>
      < Report />
    </Router>)
  })
  it('匹配快照', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('ui渲染正确， 一个img,两个p', () => {
    const emptyImg = wrapper.getByAltText('暂无主观量表');
    expect(emptyImg).toBeInTheDocument();
    expect(wrapper.container.querySelectorAll('p').length).toBe(2);
  })
  it('a链接正确', () => {
    expect(wrapper.getByTestId('link').getAttribute('href')).toMatch(/\/query/)
  });
});
