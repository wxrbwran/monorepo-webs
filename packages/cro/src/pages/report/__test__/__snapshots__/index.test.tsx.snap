// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`报告页面 匹配快照 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="report-empty"
      >
        <img
          alt="暂无主观量表"
          class="no-data-img"
          src="no_table.png"
        />
        <p>
           暂未生成报告， 
        </p>
        <p>
          请您
          <a
            data-testid="link"
            href="/query"
          >
            去查询
          </a>
          模块根据您的查询条件生成科研报告吧！
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="report-empty"
    >
      <img
        alt="暂无主观量表"
        class="no-data-img"
        src="no_table.png"
      />
      <p>
         暂未生成报告， 
      </p>
      <p>
        请您
        <a
          data-testid="link"
          href="/query"
        >
          去查询
        </a>
        模块根据您的查询条件生成科研报告吧！
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
